front/lib/plans/stripe.ts

```diff
@@ -103,3 +127,68 @@ export const getProduct = async (
   const product = await stripe.products.retrieve(productId);
   return product;
 };
+
+/**
+ * Calls the Stripe API to update the quantity of a subscription.
+ * https://stripe.com/docs/billing/subscriptions/upgrade-downgrade
+ */
+export const updateStripeSubscriptionQuantity = async ({
+  stripeSubscriptionId,
+  stripeProductId,
+  stripeCustomerId,
+  quantity,
+}: {
+  stripeSubscriptionId: string;
+  stripeProductId: string;
+  stripeCustomerId: string;
+  quantity: number;
+}): Promise<void> => {
+  // First, we get the current subscription
+  const stripeSubscriptions = await stripe.subscriptions.list({
+    customer: stripeCustomerId,
+  });
+
+  if (stripeSubscriptions.data.length !== 1) {
```

Comment:
Because that's called when a new user is onboarding on a workspace and I figure we don't want to break their sign-in but instead be notified and handle the error on our own. Does that make sense?

Reply:
Then maybe we should `catch` there ? If we want to re-use these functions elsewhere, I think it's not ideal to "fail silently"

Reply:
Done, the stripe function throws and I wrap the call in a try/catch on `updateWorkspacePerSeatSubscriptionUsage`